### PR TITLE
fix(3098): Workflow graph - Non stage jobs overlap into the stage

### PR DESCRIPTION
**Pull request by sagar1312**
## Context

Non stage jobs are being positioned under stage on the workflow graph in the UI giving a false impression that those jobs are part of the stage.

**Example (screwdriver.yaml)**
```
## Stages by grouping
shared:
    image: node:20
    steps:
        - init: echo 'init'
jobs:
    triggering-stage:
        requires: [~pr, ~commit]
    overlapping-non-stage-job-at-start-1:
        requires: [triggering-stage]
    overlapping-non-stage-job-at-start-2a:
        requires: [ overlapping-non-stage-job-at-start-1 ]
    overlapping-non-stage-job-at-start-2b:
        requires: [ overlapping-non-stage-job-at-start-1 ]
    ci-deploy:
        requires: [ ~stage@integration:setup ]
    ci-test-batch-1:
        requires: [ ci-deploy ]
    ci-test-batch-2:
        requires: [ ci-deploy ]
    ci-certify:
        requires: [ ci-test-batch-1, ci-test-batch-2 ]
    overlapping-non-stage-job-at-end-1:
        requires: [ ci-test-batch-1 ]
    overlapping-non-stage-job-at-end-2:
        requires: [ ci-test-batch-2 ]
    triggered-by-stage:
        requires: [ ~stage@integration:teardown ]

stages:
    integration:
        requires: [triggering-stage]
        jobs: [ci-deploy, ci-test-batch-1, ci-test-batch-2, ci-certify]
        description: "This stage will deploy the latest application to CI environment and certifies it after the tests are passed."
```

**Workflow graph (Before)**
image

## Objective

Non stage jobs must be positioned outside stage matrix.

**Workflow graph (After)**
image

## References

https://github.com/screwdriver-cd/screwdriver/issues/3098

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/utils/graph-tools.js [hunk 1]
```diff
@@ -305,12 +272,47 @@ const walkGraph = (graph, start, x, y) => {
...
+        // block all the columns for stage rows
+        const afterStageY = stageStartY + stageGraph.meta.height;
+
+        let stageColumn = stageStartX;
+
+        while (stageColumn <= stageEndX) {
+          y[stageColumn] = afterStageY;
+          stageColumn += 1;
+        }
+      }
```
Comment (pritamstyz4ever) [inline app/utils/graph-tools.js:306]: This logic of traversing stage nodes and applying offsets looks similar to https://github.com/screwdriver-cd/ui/pull/1045/files#diff-1e81a193b56d9ac00e3e89b9f7bb4d446837f1dea429e5534e19d64bfa4b8019R414-R446
can they be refactored into a function like drawStage/traverseStage

**Review by pritamstyz4ever — COMMENTED** [review]
minor comment

**Review by tkyi — COMMENTED** [review]
Test failure
```
17:13:46 not ok 753 Chrome 124.0 - [1 ms] - Unit | Utility | graph tools: Non stage jobs should not be positioned inside stages
17:13:46     ---
17:13:46         actual: >
17:13:46             null
17:13:46         stack: >
17:13:46             TypeError: assert.isTrue is not a function
17:13:46                 at http://localhost:7357/assets/tests.js:32868:18
17:13:46                 at Array.forEach (<anonymous>)
17:13:46                 at Object.<anonymous> (http://localhost:7357/assets/tests.js:32863:22)
17:13:46                 at runTest (http://localhost:7357/assets/chunk.vendors-node_modules_qunit_qunit_qunit_js.154011ed7183e517c3d7.js:473:197)
17:13:46                 at Test.run (http://localhost:7357/assets/chunk.vendors-node_modules_qunit_qunit_qunit_js.154011ed7183e517c3d7.js:471:830)
17:13:46                 at http://localhost:7357/assets/chunk.vendors-node_modules_qunit_qunit_qunit_js.154011ed7183e517c3d7.js:515:958
17:13:46                 at processTaskQueue (http://localhost:7357/assets/chunk.vendors-node_modules_qunit_qunit_qunit_js.154011ed7183e517c3d7.js:398:116)
17:13:46                 at http://localhost:7357/assets/chunk.vendors-node_modules_qunit_qunit_qunit_js.154011ed7183e517c3d7.js:398:178
17:13:46         message: >
17:13:46             Died on test #4: assert.isTrue is not a function
17:13:46                 at Object.<anonymous> (http://localhost:7357/assets/tests.js:32669:21)
17:13:46                 at processModule (http://localhost:7357/assets/chunk.vendors-node_modules_qunit_qunit_qunit_js.154011ed7183e517c3d7.js:272:622)
17:13:46                 at module$1 (http://localhost:7357/assets/chunk.vendors-node_modules_qunit_qunit_qunit_js.154011ed7183e517c3d7.js:277:92)
17:13:46                 at Module.callback (http://localhost:7357/assets/tests.js:31248:21)
17:13:46                 at Module.exports (http://localhost:7357/assets/vendor.js:125:32)
17:13:46                 at requireModule (http://localhost:7357/assets/vendor.js:46:18)
17:13:46                 at TestLoader.require (http://localhost:7357/assets/test-support.js:23606:9)
17:13:46         negative: >
17:13:46             false
17:13:46         browser log: |
17:13:46     ...
```

**Comment by sagar1312** [comment]
@tkyi @pritamstyz4ever 
Addressed the suggestions. Please take a look.

**Review by pritamstyz4ever — APPROVED** [review]

**Pull request merged**